test/geometry_benchmark.cpp

```diff
@@ -0,0 +1,442 @@
+/////////////////////////////////////////////////////////////////////
+//
+// Copyright (c) 2024 Matthew Fitzgerald
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//
+/////////////////////////////////////////////////////////////////////
+
+// catch2
+#include "catch2/catch_test_macros.hpp"
+#include <catch2/benchmark/catch_benchmark.hpp>
+
+// saber
+#include "saber/geometry/point.hpp"
+#include "saber/geometry/size.hpp"
+
+// std
+#include <array>
+#include <ctime>
+#include <iostream>
+#include <string>
+#include <tuple>
+#include <vector>
+
+saber::geometry::Point<float> sFloatPoint{};
+saber::geometry::Point<double> sDoublePoint{};
+saber::geometry::Point<int> sIntPoint{};
+
+TEST_CASE("saber::geometry::Point", "[saber]")
+{
+	int gauranteedNotConstexpr = 0;
+	{
+		// Get current time
```

Comment:
Add comment explaining that this block scope is to avoid compile time optimization caused by const value variables.